external/fv3fit/fv3fit/reservoir/transformers/transformer.py

```diff
@@ -20,3 +27,85 @@ def encode(self, x: Sequence[ArrayLike]) -> np.ndarray:
     @abc.abstractmethod
     def decode(self, x: ArrayLike) -> Sequence[ArrayLike]:
         pass
+
+
+@io.register("do-nothing-transformer")
+class DoNothingAutoencoder(Transformer, Reloadable):
+    _CONFIG_NAME = "mock_transformer.yaml"
+
+    """Useful class for tests. Encode just concatenates input
+    variables. Decode separates them back into individual arrays.
+    """
+
+    def __init__(
+        self, latent_dim_len, original_feature_sizes: Optional[Sequence[int]] = None
+    ):
+        self._latent_dim_len = latent_dim_len
+        self.original_feature_sizes = original_feature_sizes
+
+    @property
+    def n_latent_dims(self):
+        return self._latent_dim_len
+
+    def encode(self, x):
+        self.original_feature_sizes = [arr.shape[-1] for arr in x]
+        return np.concatenate(x, -1)
+
+    def decode(self, latent_x):
+        if self.original_feature_sizes is None:
+            raise ValueError("Must encode data before decoding.")
+
+        split_indices = np.cumsum(self.original_feature_sizes)[:-1]
+        return np.split(latent_x, split_indices, axis=-1)
+
+    def dump(self, path: str) -> None:
+        with fsspec.open(os.path.join(path, self._CONFIG_NAME), "w") as f:
+            yaml.dump(
+                {
+                    "latent_dim_len": self.n_latent_dims,
+                    "original_feature_sizes": self.original_feature_sizes,
+                },
+                f,
+            )
+
+    @classmethod
+    def load(cls, path: str) -> "DoNothingAutoencoder":
+        with fsspec.open(os.path.join(path, cls._CONFIG_NAME), "r") as f:
+            config = yaml.safe_load(f)
+        return cls(
+            latent_dim_len=config["latent_dim_len"],
+            original_feature_sizes=config["original_feature_sizes"],
+        )
+
+
+def decode_columns(
```

Comment:
I think this works since the encode/decode of the two transformers share the same call signature, but what was the intent for providing a wrapper function vs. letting the transformers perform the ops to meet the expectations?

Reply:
Since `RankDivider.merge_subdomains` already exists and operates on the raw fully flattened prediction to reshape into (x, y, latent_feature) dims, I found it easiest to have a function with a similar input data format as `encode_columns` that stacks this output with using the latent_feature dim as its unstacked dim. The alternative of reshaping the raw flat prediction array to [sample, latent_feature] dims and then just calling `Transformer.decode` on that would also work, but would require reflattening the decoded data before passing to `merge_subdomains` to get the final (x, y, z_feature) dims.